agent/qrexec-agent-data.c

```diff
@@ -152,17 +156,18 @@ void send_exit_code(libvchan_t *data_vchan, int status)
  *  1 - some data processed, call it again when buffer space and more data
  *      available
  */
-int handle_input(libvchan_t *vchan, int fd, int msg_type)
+int handle_input(libvchan_t *vchan, int fd, int msg_type, int data_protocol_version)
 {
-    char buf[MAX_DATA_CHUNK];
-    int len;
+    const size_t max_len = data_protocol_version < 3 ? MAX_DATA_CHUNK_V2 : MAX_DATA_CHUNK_V3;
+    char buf[max_len];
+    ssize_t len;
     struct msg_header hdr;
 
     hdr.type = msg_type;
     while (libvchan_buffer_space(vchan) > (int)sizeof(struct msg_header)) {
         len = libvchan_buffer_space(vchan)-sizeof(struct msg_header);
-        if (len > (int)sizeof(buf))
-            len = sizeof(buf);
+        if (len > (int)max_len)
```

Comment:
cast to int is wrong. Probably we want `(size_t)len > max_len`? (The mixing of `size_t` and `ssize_t` in functions like `read` is really annoying)